apis/tempo/v1alpha1/tempostack_webhook.go

```diff
@@ -171,43 +170,6 @@ func (v *validator) validateServiceAccount(ctx context.Context, tempo TempoStack
 	return allErrs
 }
 
-// ValidateStorageSecret validates the object storage secret required for tempo.
-func ValidateStorageSecret(tempo TempoStack, storageSecret corev1.Secret) field.ErrorList {
```

Comment:
why is this moved out of the webhook file, when its used only there?

Reply:
It is used in:
* `controllers/tempo/tempostack_controller.go`
* `apis/tempo/v1alpha1/tempostack_webhook.go`

And I move it to `apis/tempo/v1alpha1/storage.go` with all the helper functions I created to do the validation. I think it totally makes sense to avoid making the `apis/tempo/v1alpha1/tempostack_webhook.go` bigger and, at least for me, is more intuitive in this way.